src/V3ThreadPool.cpp

```diff
@@ -0,0 +1,124 @@
+// -*- mode: C++; c-file-style: "cc-mode" -*-
+//*************************************************************************
+// DESCRIPTION: Verilator: Thread pool for Verilator itself
+//
+// Code available from: https://verilator.org
+//
+//*************************************************************************
+//
+// Copyright 2005-2023 by Wilson Snyder.  This program is free software; you can
+// redistribute it and/or modify it under the terms of either the GNU
+// Lesser General Public License Version 3 or the Perl Artistic License
+// Version 2.0.
+// SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0
+//
+//*************************************************************************
+
+#include "config_build.h"
+
+#include "V3ThreadPool.h"
+
+#include "V3Error.h"
+
+void V3ThreadPool::resize(unsigned n) VL_MT_UNSAFE {
+    VerilatedLockGuard lock{m_mutex};
+    VerilatedLockGuard stoppedJobsLock{m_stoppedJobsMutex};
+    UASSERT(m_queue.empty(), "Resizing busy thread pool");
+    // Shut down old threads
+    m_shutdown = true;
+    m_stoppedJobs = 0;
+    m_cv.notify_all();
+    m_stoppedJobsCV.notify_all();
+    stoppedJobsLock.unlock();
+    lock.unlock();
+    while (!m_workers.empty()) {
+        m_workers.front().join();
+        m_workers.pop_front();
```

Comment:
pop isn't threadsafe and you don't have a lock at this point.

Reply:
This functions is marked as ``VL_MT_UNSAFE``.

Reply:
But the first thing it does is get a lock which is confusing. Please add a comment to emphasize it can race.